### ISSUE TITLE
MNIST e2e fails. Denied by Tekton admission controller.

### ISSUE THREAD

**Issue by stobias123** [kubeflow/kfp-tekton#319]
/kind bug

**What steps did you take and what happened:**
On a fresh install, trying to run through the samples to see pipeline behavior.

**Environment:**
OKD 4.5 cluster (K8s 1.19).
Installed tekton v0.14 via operator.

Running `kfp-tekton/samples/e2e-mnist/mnist.ipynb` to test functionality.

On the step where pipeline is submitted I get the below error.
```
HTTP response headers: HTTPHeaderDict({'Content-Type': 'application/json', 'Trailer': 'Grpc-Trailer-Content-Type', 'Date': 'Tue, 29 Sep 2020 06:30:25 GMT', 'Transfer-Encoding': 'chunked'})
HTTP response body: {"error":"Failed to create a new run.: InternalServerError: Failed to create a workflow for (end-to-end-pipeline-e49ca): admission webhook \"webhook.pipeline.tekton.dev\" denied the request: mutation failed: cannot decode incoming new object: json: unknown field \"metadata\"","code":13,"message":"Failed to create a new run.: InternalServerError: Failed to create a workflow for (end-to-end-pipeline-e49ca): admission webhook \"webhook.pipeline.tekton.dev\" denied the request: mutation failed: cannot decode incoming new object: json: unknown field \"metadata\"","details":[{"@type":"type.googleapis.com/api.Error","error_message":"Internal Server Error","error_details":"Failed to create a new run.: InternalServerError: Failed to create a workflow for (end-to-end-pipeline-e49ca): admission webhook \"webhook.pipeline.tekton.dev\" denied the request: mutation failed: cannot decode incoming new object: json: unknown field \"metadata\""}]}
```

**What did you expect to happen:**
I want the example to work :(

**Comment by Tomcli** [comment]
Related issue: https://github.com/tektoncd/pipeline/issues/2830
@stobias123 you need to uninstall the OKD 4.5 openshift pipeline then install kubeflow.

**Comment by stobias123** [comment]
@Tomcli  - After further inspection..  I have not installed openshift pipelines, nor tekton via the operator. I've updated the image as such.

Inspecting my kfdef shows that I'm deploying the below as my tekton deployment... 

https://github.com/adrian555/manifests/blob/update-manifests-repo-link/tektoncd/tektoncd-install/base/deployment.yaml

**Comment by Tomcli** [comment]
@stobias123 Correct me if I'm wrong. Isn't OKD 4.5 cluster comes with openshift pipelines operator by default?

Also, did you use the option 1 in this deployment doc? 
https://github.com/IBM/KubeflowDojo/tree/master/OpenShift/manifests#1-leverage-openshift-pipelines

**Comment by Tomcli** [comment]
You can run `oc get pod -n openshift-operators` and see is the openshift pipelines operator deployed on your machine.

**Comment by stobias123** [comment]
@Tomcli  - Re: OKD pipelines. No it's not pre-installed. And I literally mean OKD, not OCP, so I dont even get access to RH supported operators.

**Comment by Tomcli** [comment]
@stobias123 sorry I mistaken the OKD abbreviation. The error above usually happens when someone try to overwrite an existing Tekton CRDs. Just to verify, is this error also show up with the flip-coin example from the default kfp-tekton deployment?

**Comment by Tomcli** [comment]
Also just to check with the python kernel inside the jupyter notebook. can you run this python code in the jupyter notebook to see the pipelinerun spec?
```python
from kfp_tekton.compiler import TektonCompiler
TektonCompiler().compile(mnist_pipeline, 'mnist-pipeline.yaml')
```

**Comment by stobias123** [comment]
I destroyed custer and re-created... things look ok - flip-coin pipeline worked. I appreciate the quick responses, but I'll close, this was installation issues due to user error 🤷 

I'll leave the below comments in case someone comes across in the future.

1. I was wrong in above comments about my install method. 
2. It's very unclear what's actually in the min install, I can't even get notebooks or view pipeline ui (I get response from pipeline ui but nothing shows)
3. Ultimately I ended up a) installing min yaml then b) installing the kfserving yaml.

I'll chalk this up to user error for the most part, but it's unfortunate how challenging it is to remove kubeflow and start over. I've always had to blast a cluster. :(
